### DISCUSSION TITLE
Missing diagnostics for erroneous STL container's emplace_back

### DISCUSSION THREAD

**Discussion by Ceron257** [clangd/clangd#1272] [Q&A] [answered]
Consider the following code:
```cpp
#include <vector>

class Bar
{};

class Foo
{
public:
  Foo (Bar b)
    : bar (b)
  {}

private:
  Bar bar;
};

int main ()
{
  std::vector<Foo> test;
  test.emplace_back (0);
  return 0;
}
```
It doesn't build, but clangd doesn't show a diagnostic that indicates this. Would it be possible to get this feedback from clangd without the need to actually build the code? If not what are the reasons for that?

**Comment by HighCommander4** [comment, marked as answer]
This is discussed in #137
